libusb/os/linux_usbfs.c

```diff
@@ -413,6 +420,23 @@ static void op_exit(struct libusb_context *ctx)
 	usbi_mutex_static_unlock(&linux_hotplug_startstop_lock);
 }
 
+static int op_set_option(struct libusb_context *ctx, enum libusb_option option, va_list ap)
+{
+	UNUSED(ctx);
+	UNUSED(ap);
+
+	switch ((int)option) {
```

Comment:
Why is there a cast here?  Is it to avoid a compiler warning, or is it copied from windows_common.c?

Reply:
Copied from op_exit to avoid a compiler warning.

Reply:
What is the warning?

Reply:
jni/../../libusb/os/linux_usbfs.c:427:49: warning: unused parameter 'ctx' [-Wunused-parameter]
static int op_set_option(struct libusb_context *ctx, enum libusb_option option, va_list ap)
                                                ^
jni/../../libusb/os/linux_usbfs.c:427:89: warning: unused parameter 'ap' [-Wunused-parameter]
static int op_set_option(struct libusb_context *ctx, enum libusb_option option, va_list ap)

Reply:
I think the question was about the cast in `(int)option`

Reply:
Exactly. Casting to an int for the switch is odd.

Reply:
Sorry i misunderstood. Thank LudovicRousseau. The cast `(int)option` is just copied from windows_common.c.

Reply:
I have removed the cast in commit `86997c8`.